src/main.cpp

```diff
@@ -2086,14 +2086,16 @@ static bool ActivateBestChainStep(CValidationState &state, CBlockIndex *pindexMo
 // or an activated best chain. pblock is either NULL or a pointer to a block
 // that is already loaded (to avoid loading it again from disk).
 bool ActivateBestChain(CValidationState &state, CBlock *pblock) {
-    CBlockIndex *pindexNewTip = NULL;
     CBlockIndex *pindexMostWork = NULL;
     do {
         boost::this_thread::interruption_point();
 
+        CBlockIndex *pindexNewTip;
```

Comment:
I'm not sure I agree with this as best practice (initializing variables may hide bugs too that uninitialized memory and eg valgrind could show), but: done.